src/ai/backend/manager/models/audit_logs.py

```diff
@@ -0,0 +1,266 @@
+from __future__ import annotations
+
+import enum
+import logging
+from typing import TYPE_CHECKING, Optional, Sequence
+
+import graphene
+import sqlalchemy as sa
+from dateutil.parser import parse as dtparse
+from graphene.types.datetime import DateTime as GQLDateTime
+from sqlalchemy.dialects import postgresql as pgsql
+from sqlalchemy.engine.row import Row
+
+from ai.backend.common.logging import BraceStyleAdapter
+
+from .base import EnumValueType, Item, PaginatedList, metadata, simple_db_mutate
+from .minilang.ordering import QueryOrderParser
+from .minilang.queryfilter import QueryFilterParser
+from .user import UserRole
+
+if TYPE_CHECKING:
+    from .gql import GraphQueryContext
+log = BraceStyleAdapter(logging.getLogger(__name__))
+__all__: Sequence[str] = (
+    "audit_logs",
+    "AuditLog",
+    "AuditLogList",
+    "AuditLogInput",
+    "CreateAuditLog",
+)
+
+
+class AuditLogAction(str, enum.Enum):
+    """AuditLog Action's Enum"""
+
+    CREATE = "CREATE"
+    CHANGE = "CHANGE"
+    DELETE = "DELETE"
+
+
+class AuditLogTargetType(str, enum.Enum):
+    """ """
+
+    USER = "user"
+    KEYS = "keypairs"
+    VFOLDER = "vfolder"
```

Comment:
It is quite annoying to add enum value to existing Enum type in DB through alembic.
I recommend you to add all available target types here and update alembic migration file to reduce future annoying work!
Plus, please unify the name in singular or plural.

```suggestion
class AuditLogTargetType(str, enum.Enum):
    """ """

    USER = "user"
    GROUP = "group"
    KEYPAIRS = "keypair"
    VFOLDER = "vfolder"
    COMPUTE_SESSION = "compute_session"
```